documentation/examples/gh-got.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const got = require('../..');
+const package = require('../../package');
+
+const getRateLimit = ({headers}) => ({
+	limit: parseInt(headers['x-ratelimit-limit'], 10),
+	remaining: parseInt(headers['x-ratelimit-remaining'], 10),
+	reset: new Date(parseInt(headers['x-ratelimit-reset'], 10) * 1000)
+});
+
+const instance = got.extend({
+	baseUrl: 'https://api.github.com',
+	headers: {
+		accept: 'application/vnd.github.v3+json',
+		'user-agent': `${package.name}/${package.version}`
+	},
+	responseType: 'json',
+	token: process.env.GITHUB_TOKEN,
+	handlers: [
+		(options, next) => {
+			// Authorization
+			if (options.token && !options.headers.authorization) {
+				options.headers.authorization = `token ${options.token}`;
+			}
+
+			// Don't touch streams
+			if (options.stream) {
+				return next(options);
+			}
+
+			// Magic begins.
+			return (async () => {
+				try {
+					const response = await next(options);
```

Comment:
```suggestion
					const promiseOrStream = await next(options);
```

Reply:
This is neither a response nor stream. There is a check that returns `next(options)` if `options.stream` is true. Yup, it would be a promise but we need to use `await` to get the response.